core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java

```diff
@@ -168,11 +168,24 @@ public static LoadTableResponse stageTableCreate(Catalog catalog, Namespace name
     properties.put("created-at", OffsetDateTime.now().toString());
     properties.putAll(request.properties());
 
+    String location;
+    if (request.location() != null) {
+      location = request.location();
+    } else {
+      location = catalog.buildTable(ident, request.schema())
```

Comment:
A missing location returned by `stageCreate` was causing `TableMetadata` deserialization to fail. This delegates creating a table location to the wrapped catalog.